### PR TITLE
KOGITO-3330 Hotfix: Conditional eager reading of resources on native-image

### PR DESCRIPTION
**Pull request by evacchi**
https://issues.redhat.com/browse/KOGITO-3330

In GraalVM 20.2 under certain conditions, it is not possible to hold onto an input stream reader that has been generated from a classpath resource -- see https://github.com/quarkusio/quarkus/issues/8694#issuecomment-691828808

The hotfix consists in eagerly reading the entire resource when native mode is detected at build time (detected via special System property).

in other words, instead of:

```
new InputStreamReader(...getResourceAsStream())
```

for native image we do 

```
bytes = getResourceAsStream().readAllBytes()
new InputStreamReader(new ByteArrayInputStream(bytes))
```

We may have to revisit the issue to see if there are "better" ways to fix it (e.g. Quarkus has most probably a "proper" way to handle resources); in the meantime, we need to merge this to proceed with the release today.

(tested locally with native-image. You can double check integration-tests/integration-tests-quarkus  if you want)

### CODE REVIEW

File: kogito-codegen/src/main/resources/class-templates/DecisionContainerTemplate.java [hunk 1]
```diff
@@ -1,9 +1,7 @@
-import java.util.function.Function;
-
-import org.kie.api.runtime.KieRuntimeFactory;
-
 public class DecisionModels implements org.kie.kogito.decision.DecisionModels {
 
+    private final static boolean IS_JDK = System.getProperty("org.graalvm.nativeimage.imagecode") == null;
```
Comment (geoand) [inline kogito-codegen/src/main/resources/class-templates/DecisionContainerTemplate.java:3]: I would personally invert this and call it `IS_GRAALVM` :)

Comment (evacchi) [reply kogito-codegen/src/main/resources/class-templates/DecisionContainerTemplate.java:3]: let's call it `IS_NATIVE_IMAGE` then

**Review by geoand — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

File: kogito-codegen/src/main/resources/class-templates/DecisionContainerTemplate.java [hunk 2]
```diff
@@ -1,9 +1,7 @@
-import java.util.function.Function;
-
-import org.kie.api.runtime.KieRuntimeFactory;
-
 public class DecisionModels implements org.kie.kogito.decision.DecisionModels {
 
+    private final static boolean IS_NATIVE_IMAGE = System.getProperty("org.graalvm.nativeimage.imagecode") != null;
```
Comment (tarilabs) [inline kogito-codegen/src/main/resources/class-templates/DecisionContainerTemplate.java:3]: Good for me, I had similar to 
https://github.com/kiegroup/drools/blob/d72e9594cbb7f71a70f1bc13cf84b168ffa52e69/kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/ClassLoaderUtil.java#L21
eventually I can consolidate

**Review by tarilabs — APPROVED** [review]

File: kogito-codegen/src/main/resources/class-templates/DecisionContainerTemplate.java (hunk 2 above)
Comment (evacchi) [reply kogito-codegen/src/main/resources/class-templates/DecisionContainerTemplate.java:3]: ok let's open a JIRA

Comment (evacchi) [reply kogito-codegen/src/main/resources/class-templates/DecisionContainerTemplate.java:3]: https://issues.redhat.com/browse/KOGITO-3331

**Review by danielezonca — APPROVED** [review]
LGTM

**Pull request merged**